src/connect/components/tree-view/tree-view.tsx

```diff
@@ -6,78 +6,88 @@ import {
+    onDragEnd?: ConnectTreeViewItemProps['onDragEnd'];
     disableHighlightStyles?: boolean;
+    filterPath?: string;
+    level?: number;
+    allowedPaths?: string[];
 }
 
-export function ConnectTreeView<T extends string = string>(
-    props: ConnectTreeViewProps<T>,
-) {
+export function ConnectTreeView(props: ConnectTreeViewProps) {
     const {
-        items,
         onItemClick,
         onDropEvent,
         defaultItemOptions,
         onItemOptionsClick,
         onSubmitInput = () => {},
         onCancelInput = () => {},
+        filterPath,
+        level = 0,
+        allowedPaths,
         ...elementProps
     } = props;
 
-    function renderTreeItems(item: TreeItem<T>, level = 0) {
-        if (
-            item.action === ActionType.New ||
-            item.action === ActionType.Update
-        ) {
-            return (
-                <ConnectTreeViewInput
-                    item={item}
-                    key={item.id}
-                    level={level}
-                    onSubmit={onSubmitInput}
-                    onCancel={onCancelInput}
-                />
-            );
-        }
+    const items = usePathContent(filterPath, allowedPaths);
 
-        return (
-            <ConnectTreeViewItem<T>
-                item={item}
-                key={item.id}
-                onDropEvent={onDropEvent}
-                onOptionsClick={onItemOptionsClick}
-                defaultOptions={defaultItemOptions}
-                onClick={e => onItemClick?.(e, item)}
-                disableDropBetween={level === 0 && !item.expanded}
-                {...elementProps}
-            >
-                {item.children?.map(item => renderTreeItems(item, level + 1))}
-            </ConnectTreeViewItem>
-        );
-    }
+    return (
+        <>
+            {items.map((item) => {
+                if (item.action === ActionType.New ||
+                    item.action === ActionType.Update) {
+                    return (
+                        <ConnectTreeViewInput
+                            item={item}
+                            key={item.id}
+                            level={level}
+                            onSubmit={onSubmitInput}
+                            onCancel={onCancelInput}
+                        />
+                    );
+                }
 
-    return renderTreeItems(items);
+                return (
+                    <ConnectTreeViewItem
+                        item={item}
+                        level={level}
+                        key={item.id}
+                        onDropEvent={onDropEvent}
+                        onOptionsClick={onItemOptionsClick}
+                        defaultOptions={defaultItemOptions}
+                        onClick={e => onItemClick?.(e, item)}
+                        disableDropBetween={level === 0 && !item.expanded}
+                        {...elementProps}
+                    >
+                        {item.expanded && (
```

Comment:
this avoid performing path filter items for those folders that are not open. this should help us to improve the performance